qt/widgets/instrumentview/src/XIntegrationControl.cpp

```diff
@@ -145,10 +147,47 @@ bool XIntegrationScrollBar::eventFilter(QObject *object, QEvent *e) {
       break;
 
     case Qt::Key_Right:
-      if (sliderx + slider->width() + step < totalWidth) {
+      if (sliderx + sliderWidth + step < totalWidth) {
         m_slider->move(sliderx + step, slidery);
       } else {
-        m_slider->move(totalWidth - slider->width(), slidery);
+        m_slider->move(totalWidth - sliderWidth, slidery);
+      }
+      m_changed = true;
+      updateMinMax();
+      break;
+
+    case Qt::Key_Up:
+      if (sliderWidth + step <= totalWidth) {
+        if (sliderx > step) {
```

Comment:
All these if statements are a bit hard to look through, can you put in some comments on what each catch is checking for?

Reply:
Well, they were so confusing, one of them was actually redundant. It should be easier to understand now.